### ISSUE TITLE
Default buffer quad_segs is different

### ISSUE THREAD

**Issue by mwtoews** [shapely/shapely#2056]
There are two different defaults for `quad_segs` parameters for shapely's buffer operation:

- [`object.buffer(..., quad_segs=16, ...)`](https://shapely.readthedocs.io/en/2.0.4/manual.html#object.buffer) which was the default for shapely since the first release
- [`shapely.buffer(..., quad_segs=8, ...)`](https://shapely.readthedocs.io/en/2.0.4/reference/shapely.buffer.html), same as [`pygeos.constructive.buffer(..., quadsegs=8, ...)`](https://pygeos.readthedocs.io/en/0.14/constructive.html#pygeos.constructive.buffer) where it originated

This difference may cause confusion, since the two results using defaults are different:
```python
import shapely
geom = shapely.geometry.Point(0.0, 0.0)
assert geom.buffer(1.0) == shapely.buffer(geom, 1.0) is False
assert geom.buffer(1.0) == shapely.buffer(geom, 1.0, quad_segs=16)
```
Suggestions:

- Change constructive `shapely.buffer` method to use same `quad_segs=16` default, which could potentially raise downstream confusion
- Keep different defaults, and document these differences

I'm surprised this didn't come up with #1512

**Comment by jorisvandenbossche** [comment]
Yeah, this difference originates from PyGEOS. I assume we followed the PostGIS default at the time (https://postgis.net/docs/ST_Buffer.html)

**Comment by mwtoews** [comment]
GEOS also has a default of 8 ([source](https://github.com/libgeos/geos/blob/4d26df6ea4cab2bd8f5cc9da7efad458267362da/include/geos/operation/buffer/BufferParameters.h#L94)).
